src/VisualStudio/Core/Def/Implementation/ProjectSystem/FileChangeTracker.cs

```diff
@@ -88,7 +89,7 @@ public void AssertUnsubscription()
         public void EnsureSubscription()
```

Comment:
💭 Seems like this should be asynchronous now to avoid blocking thread pool threads

Reply:
The only call to this at this point comes though the compiler and the call is already synchronous. And we're hoping to kill this code ASAP too, so not really much we can do here given our time remaining.

Reply:
I guess to clarify: the request from the compiler for metadata is not an async call path, so we have no choice but to block in the end.